src/Corax/Utils/EntryIdEncodings.cs

```diff
@@ -128,8 +128,8 @@ public static void DecodeAndDiscardFrequencyClassic(Span<long> entries, int read
     [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
     public static void DecodeAndDiscardFrequency(Span<long> entries, int read)
     {
-        if (Avx2.IsSupported)
-            DecodeAndDiscardFrequencyAvx2(entries, read);
+        if (Vector256.IsHardwareAccelerated)
+            DecodeAndDiscardFrequencyVector256(entries, read);
         else if (AdvSimd.IsSupported)
             DecodeAndDiscardFrequencyNeon(entries, read);
```

Comment:
Is this needed now, or is the `Vector256` generating the right instructions in ARM as well so we don't need specialized code?

Reply:
That's for the case we have `AdvSimd` instruction set but not the whole necessary to support `Vector256` completely (it could technically happen).